test/java/src/test/java/TestJsonSerializer.java

```diff
@@ -0,0 +1,104 @@
+import static org.junit.Assert.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.math.BigDecimal;
+import java.util.HashSet;
+
+import com.fasterxml.jackson.core.*;
+
+import org.junit.Test;
+
+import org.joda.time.LocalDate;
+
+import io.sysl.reljam.gen.tuple.complex.BuyItemFormComplex;
+import io.sysl.reljam.gen.tuple.complex.CustomerType;
+import io.sysl.reljam.gen.tuple.complex.Email;
+import io.sysl.reljam.gen.tuple.complex.UserFormComplexJsonDeserializer;
+import io.sysl.reljam.gen.tuple.complex.UserFormComplexJsonSerializer;
+
+public class TestJsonSerializer
+{
+    @Test
+    public void testSerializeDeserialize() throws IOException {
+        BuyItemFormComplex e1 = loadBuyItemFormComplex("src/main/resources/test_1.json");
+        BuyItemFormComplex e2 = createBuyItemFormComplex();
+        assertEquals(e1, e2);
+
+        String s1 = serialize(e1);
+        String s2 = serialize(e2);
+        assertEquals(s1, s2);
+    }
+
+    @Test
+    public void testSubObject()  throws IOException {
+        BuyItemFormComplex entity = loadBuyItemFormComplex("src/main/resources/test_2.json");
+        String str = serialize(entity);
+        assertTrue(str.equals(WITH_SUB_OBJECT));
```

Comment:
If you insist on the string comparison - please explain the benefits (whitespace makes this more brittle) and rename `WITH_SUB_OBJECT` to `TEST2_JSON`.

Reply:
the benefit is you don't need to add any new files. Have a look at TestPetShop.java.